### ISSUE TITLE
Non-english language

### ISSUE THREAD

**Issue by Tanshu334** [tinyfish-io/tf-playwright-stealth#28]
Because of the user-agent, the website serves a non english language to the browser which breaks my code

**Comment by yungd1plomat** [comment]
just use selectors based on reliable selectors like class names, ids and attributes

**Comment by Tanshu334** [comment]
> just use selectors based on reliable attributes like class names, ids and etc
no, i can't really do that. i'm making an AI recaptcha solver, the text on the recaptcha is in some random language which breaks my code. before applying the stealth it's in english so it is because of the stealth.

**Comment by gianpaj** [comment]
It looks like it depends on `_generate_accept_language()` and since `domain_code="com"` is hard coded  in `_properties.py`

```py
spoofed_headers = FakeHttpHeader(domain_code="com", browser=browser_type.value)
```

it will randomly pick a country from `COUNTRY_TOP_LEVEL_DOMAINS` in `constants.py` in `fake-http-header`

https://github.com/MichaelTatarski/fake-http-header/blob/c684b6e1fca568b6a9794aea4cd5e8eb91c7c940/fake_http_header/util.py#L40-L41

https://github.com/tinyfish-io/tf-playwright-stealth/blob/71fe7cf097fee3961344ecb75c549290e860f520/playwright_stealth/properties/_properties.py#L28

If you set it to `us` you have a higher chance to get an English locale, but it's still random between
```
'us': ['en-US', 'es-US', 'haw', 'fr']
```

I think this is something that should be able to overwrite as this **doesn't** work:
```py
page = browser.new_page()
config = StealthConfig(languages=("en-US", "en"))
stealth_sync(page, config=config)
page.goto("https://bing.com")
```

**Comment by VeArnold** [comment]
I struggled with changing the language too, and when I dived into the code, it seemed like none of the options (except the true/false options to run specific scripts) did anything - seems like someone had started to do something with it, but left it halfway through.

Due to the `FakeHttpHeader` implementation and intertwinement in code, it's a bit difficult to "override" some values in a clean manner. The easiest approach is to just write over all the automatically generated values, but I honestly don't think it's the cleanest, but it's what I did here:
https://github.com/VeArnold/tf-playwright-stealth/commit/69d5b717868ce9dbb45950f0e57c836b6a565839

Seems like it might require quite a big chunk of refactoring to have a clean implementation. The main thing to consider is keeping the headers and js in sync (e.g. if you straight-up just write over the language values with the current implementation, then you'll have headers that are out of sync with your js).

I think it would make sense to allow providing your headers instead of the `FakeHttpHeader` and have the code still handle the rest of it - just becomes a bit messy as well if you want to add more "overrides" etc - which is why I think a major refactoring makes most sense with customisability in mind.

**Comment by Tanshu334** [comment]
this library itself seems abandoned and not maintained properly to begin with. i've switched to using patchright anyway.

**Comment by VeArnold** [comment]
Yeah, seems that way. :) This fork seems to be a bit more customizable: https://github.com/Mattwmaster58/playwright_stealth

In any case, hope someone gives more love to either one of them.
